package/src/components/MessageList/MessageList.tsx

```diff
@@ -540,13 +538,24 @@ const MessageListWithContext = <
     if (!channel || (!channel.initialized && !channel.offlineMode)) return null;
 
     const lastRead = channel.lastRead();
+    const countUnread = channel.countUnread();
 
     const lastMessage = messageList?.[index + 1];
 
+    const isLatestMessageSetShown = channel.state.messageSets.find(
+      (set) => set.isCurrent && set.isLatest,
+    );
+
+    const isMessageIndexUnread = (i: number) => isLatestMessageSetShown && i <= countUnread - 1;
+    const isUnreadMessage =
+      isMessageIndexUnread(index) || !!isUnreadMessageRef.current(message, lastRead);
+    const isLastMessageUnread =
+      isMessageIndexUnread(index + 1) || !!isUnreadMessageRef.current(lastMessage, lastRead);
+
     const showUnreadUnderlay =
-      !!isUnreadMessageRef.current(message, lastRead) && scrollToBottomButtonVisible;
-    const insertInlineUnreadIndicator =
-      showUnreadUnderlay && !isUnreadMessageRef.current(lastMessage, lastRead);
+      isUnreadMessage ||
```

Comment:
Any chance we can simplify this complex ternary? It is difficult to read and understand.

Reply:
done.. thanks for the suggestion